ui/components/src/TimeSeriesTooltip/focused-series.ts

```diff
@@ -52,17 +53,13 @@ export function getNearbySeries(
     return currentFocusedData;
   }
 
+  const focusedSeriesIndexes: number[] = [];
+  const emphasizedSeriesIndexes: number[] = [];
+  const nonEmphasizedSeriesIndexes: number[] = [];
+
   if (Array.isArray(data.xAxis) && Array.isArray(data.timeSeries)) {
     for (let seriesIdx = 0; seriesIdx < data.timeSeries.length; seriesIdx++) {
       const currentSeries = data.timeSeries[seriesIdx];
-      // TODO: look into using batch or excludeSeriesId within downplay action to fix flicker
-      if (chart?.dispatchAction !== undefined) {
-        // clears emphasis state of lines that are not focused
-        chart.dispatchAction({
-          type: 'downplay',
-          seriesIndex: seriesIdx,
-        });
-      }
```

Comment:
There is still flicker but changing from dispatching separate actions inside of the for loop and instead doing [a single chart.dispatchAction](https://github.com/perses/perses/blob/a4afe734a0e548f6525cbfa8c53da53618c485ae/ui/components/src/TimeSeriesTooltip/nearby-series.ts#L128-L132) for 'highlight' with the entire `emphasizedSeriesIndexes` array at once really helps. 

This was a big problem in the last PR #1181 since I didn't realize you could pass an array that way. Now `blur` in [getLineSeries](https://github.com/perses/perses/blob/a4afe734a0e548f6525cbfa8c53da53618c485ae/ui/panels-plugin/src/plugins/time-series-chart/utils/data-transform.ts#L96-L108) also works properly.

Long term I'm planning to dig more into this as part of the TimeChart.tsx refactor: https://github.com/perses/perses/pull/1134

In that branch using ECharts' 'time' axis, you can also define an "xBuffer" param which will potentially make the experience when moving your cursor horizontally a bit more smooth